test/itkTransformToStrainFilterTest.cxx

```diff
@@ -22,33 +22,70 @@
 #include "itkImageFileWriter.h"
 #include "itkTransformToDisplacementFieldFilter.h"
 #include "itkStrainImageFilter.h"
+#include "itkTestingMacros.h"
 
 int itkTransformToStrainFilterTest( int argc, char * argv [] )
 {
-  /** Check command line arguments. */
+  // Check command line arguments.
   if( argc < 6 )
     {
-    std::cerr << "Usage: ";
-    std::cerr << argv[0] << "<strainForm> <transformName> <strainFieldFileName> <displacementField> <displacementFieldStrain> [bSplineParametersFile]" << std::endl;
+    std::cerr << "Missing parameters." << std::endl;
+    std::cerr << "Usage: " << argv[0];
+    std::cerr << "strainForm"
+      << " transformName"
+      << " strainFieldFileName"
+      << " displacementField"
+      << " displacementFieldStrain"
+      << " [bSplineParametersFile]" << std::endl;
     return EXIT_FAILURE;
     }
 
 
-  int strainForm = 0;
-  if( ! strcmp( argv[1], "INFINITESIMAL" ) )
+  // Typedefs.
+  const unsigned int  Dimension = 2;
+  typedef float       ScalarPixelType;
+  typedef double      CoordRepresentationType;
+  const unsigned int  SplineOrder = 3;
+
+  typedef itk::Transform< CoordRepresentationType, Dimension, Dimension >
+    TransformType;
+  typedef TransformType::ParametersType ParametersType;
+
+  typedef itk::TransformToStrainFilter< TransformType, ScalarPixelType, ScalarPixelType >
+    TransformToStrainFilterType;
+
+  TransformToStrainFilterType::Pointer transformToStrainFilter =
+    TransformToStrainFilterType::New();
+
+  EXERCISE_BASIC_OBJECT_METHODS( transformToStrainFilter,
+    TransformToStrainFilter, GenerateImageSource );
+
+
+  // Test the unknown strain form exception
+  TransformToStrainFilterType::StrainFormType strainForm =
+    static_cast< TransformToStrainFilterType::StrainFormType >( -1 );
+
+  transformToStrainFilter->SetStrainForm( strainForm );
+
+  TRY_EXPECT_EXCEPTION( transformToStrainFilter->Update() );
+
+
+  // Get the input strain form
+  if( !strcmp( argv[1], "INFINITESIMAL" ) )
     {
-    strainForm = 0;
+    strainForm = TransformToStrainFilterType::StrainFormType::INFINITESIMAL;
```

Comment:
Could these please still be kept as `0`, `1`, `2` -- the compiler can complain or error if C++11 is not enabled.